### ISSUE TITLE
How to Build "coexisting-angular-microfrontends"?

### ISSUE THREAD

**Issue by ajaykumarsana** [joeldenning/coexisting-angular-microfrontends#51]
Hi @joeldenning ,

last 2 weeks i was going through your repositories, i wanted to appreciate your efforts for **Single spa**. Though i could understand the things had a doubt i.e
How to build  [this](https://github.com/joeldenning/coexisting-angular-microfrontends) repository, Is build command should be ran on each folder?

I ran **npm run build:single-spa:<app_name>** commands in each folder but that's not the solution, Could you please help me to get it done? And how to use import override in build time?

Thanks a lot

**Comment by joeldenning** [comment]
Run `npm run build:single-spa` in each of the angular directories. See code below as an example:

https://github.com/joeldenning/coexisting-angular-microfrontends/blob/925dd8ec4f7f149eeba305f4e3cc4e9783abf3db/.travis.yml#L12-L15

**Comment by ajaykumarsana** [comment]
HI @joeldenning ,

Above worked for me, Thanks.

I've another issue which is as below.
I've added [AngularJS Repo](https://github.com/joeldenning/single-spa-es5-angularjs) to [this](https://github.com/joeldenning/coexisting-angular-microfrontends) and provided extra route for Angular JS repo.
i ran http-server in [Angular repo](https://github.com/joeldenning/single-spa-es5-angularjs) then added http://localhost:8081/es5-app.js to import override when i access **Angular JS repo** route it gives CORS issue.

![image](https://user-images.githubusercontent.com/1854019/85722863-56b0ab80-b710-11ea-9583-f7271e976325.png)

**Comment by joeldenning** [comment]
`http-server --cors`

**Comment by joeldenning** [comment]
https://www.npmjs.com/package/http-server#available-options

**Comment by ajaykumarsana** [comment]
Hi @joeldenning 

I've tried following.

1) Ran `http-server --cors ` inside [AngularJS Repo](https://github.com/joeldenning/single-spa-es5-angularjs)
2) Ran `http-server --cors ` inside both  [AngularJS Repo](https://github.com/joeldenning/single-spa-es5-angularjs) and [this](https://github.com/joeldenning/coexisting-angular-microfrontends)

Still it gives Error, :Access to script at 'http://localhost:8080/es5-app.js' from origin 'http://localhost:4200' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

**Comment by ajaykumarsana** [comment]
@joeldenning 
Another issue:
I used below code to register 
`System.register('**angjs**',[], function(_export) {
  return {
    execute: function() {
      _export(singleSpaAngularjs.default({
        angular: angular,
        mainAngularModule: 'main-module',
        uiRouter: true,
        preserveGlobal: false,
        template: '<root />',
      }))
    }
  }
})`

I got [error](https://single-spa.js.org/error/?code=37&arg=application&arg=Angjs17&arg={) with **37** as code, which says valid **unmount** not exist.

**Comment by joeldenning** [comment]
I have used `http-server --cors` for years and it works. My guess is that you're loading from a different server instead of http-server, such as `ng serve` or something else. The next step I'd take to debug is to check whether the URL is correct and whether a response header Access-Control-Allow-Origin is being sent by checking the Network tab of the browser devtools.

Regarding registering the angularjs application, I recommend not doing a "named register," but instead an anonymous register.

```js
// Don't do this
System.register('angjs', [], function() {})

// Do this instead
System.register([], function () {})
```

Then add `angjs` to your import map and run the following in the browser console:

```js
System.import('angjs').then(ns => {
  console.log('angjs module', ns)
})
```

**Comment by ajaykumarsana** [comment]
@joeldenning ,
Worked for me.

Thanks
